src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,404 @@
+        {
+            return _registryManager.RegsitryExists(sid);
+        }
+
+        public void UpdateStandardRegistrySettings()
+        {
+            foreach(var regSetting in _standardRegistries)
+            {
+                SetRegistryKeyValue(regSetting.Key, regSetting.Value);
+            }
+        }
+
+        public void RevertOriginalRegistrySettings()
+        {
+            foreach(var regSetting in _standardRegistries)
+            {
+                RevertOriginalRegistry(regSetting.Key);
+            }
+
+            //auto-logon
+            RevertOriginalRegistry(WellKnownRegistries.AutoLogonUserName);
+            RevertOriginalRegistry(WellKnownRegistries.AutoLogonDomainName);
+            RevertOriginalRegistry(WellKnownRegistries.AutoLogonPassword);
+            RevertOriginalRegistry(WellKnownRegistries.AutoLogonCount);
+            RevertOriginalRegistry(WellKnownRegistries.AutoLogon);
+
+            //startup process
+            RevertOriginalRegistry(WellKnownRegistries.StartupProcess);
+        }
+
+        public void UpdateAutoLogonSettings(string userName, string domainName)
+        {
+            SetRegistryKeyValue(WellKnownRegistries.AutoLogonUserName, userName);
+            SetRegistryKeyValue(WellKnownRegistries.AutoLogonDomainName, domainName);
+
+            //this call is to take the backup of the password key if already exists as we delete the key in the next step
+            SetRegistryKeyValue(WellKnownRegistries.AutoLogonPassword, "");
+            DeleteRegistry(WellKnownRegistries.AutoLogonPassword);
+
+            //this call is to take the backup of the password key if already exists as we delete the key in the next step
+            SetRegistryKeyValue(WellKnownRegistries.AutoLogonCount, "");
+            DeleteRegistry(WellKnownRegistries.AutoLogonCount);
+
+            SetRegistryKeyValue(WellKnownRegistries.AutoLogon, "1");
+        }
+
+        public string GetAutoLogonUserName()
+        {
+            var regValue = GetRegistryKeyValue(WellKnownRegistries.AutoLogon);
+            int.TryParse(regValue, out int autoLogonEnabled);
+
+            return autoLogonEnabled == 1
+                    ? GetRegistryKeyValue(WellKnownRegistries.AutoLogonUserName)
+                    : null;
+        }
+
+        public void SetStartupProcessCommand(string startupCommand)
+        {            
+            SetRegistryKeyValue(WellKnownRegistries.StartupProcess, startupCommand);
+        }
+
+        public string GetStartupProcessCommand()
+        {            
+            return GetRegistryKeyValue(WellKnownRegistries.StartupProcess);
+        }
+
+        public List<string> GetAutoLogonRelatedWarningsIfAny()
+        {
+            var warningReasons = new List<string>();
+
+            //screen saver
+            var screenSaverValue = GetRegistryKeyValue(WellKnownRegistries.ScreenSaverDomainPolicy);
+            int.TryParse(screenSaverValue, out int isScreenSaverDomainPolicySet);
```

Comment:
the int will have its default value and I am specifically looking for '1'.

---
In reply to: [118025666](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118025666) [](ancestors = 118025666)